File: smarthome.FrontEnd/Smarthome/src/app/models/Notification.ts
import { HomeMember } from "./home";

export type Notification = {
    notificationId: number;
    eventName: string;
    hardwareId: number;
    createdAt: string;
}

export type HomeMemberNotification = {
    notificationId: number;
    notification: Notification; 
    homeId: number;
    userId: number;
    homeMember: HomeMember;
    read: boolean;
  };
  

File: smarthome.Domain/Notification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace smarthome.Domain;

public class Notification
{
    [Key]
    public int NotificationId { get; set; }

    [Required]
    public string EventName { get; set; }

    [Required]
    [ForeignKey(nameof(HomeDevice))]
    public int HardwareId { get; set; }

    [Required]
    public string CreatedAt { get; set; }
}
